### PR TITLE
Fix Xcode Index Store Resource Starvation Race

### PR DESCRIPTION
**Pull request by jerrymarino**
Xcode watches the indexer store for changes. If this directory is populated
during a build, Xcode will starve the build process for usage.

This patch uses 2 index stores: 1 for build time and 1 for after, which Xcode
"knows about". After a build, we sync the Tulsi index store with the Xcode one.

From Xcode's perspective, the files are still put in the same place, just in
a batch rather than incrementally.

Change-Id: I7cdd2476172e6bcee435ebf9a7e33d7e3b09b7e9

### CODE REVIEW

**Comment by jerrymarino** [comment]
cc @rahul-malik @bkase  I found this issue to tank the build time. With the index store feature added, I saw another 3-4 minutes on average. This seems to fix it!

**Review by bkase — APPROVED** [review]
Can reproduce it working. Ship it

**Comment by jerrymarino** [comment]
Thanks @bkase - glad it's working for you man!

For the record, this was originally added here: https://github.com/pinterest/tulsi/pull/17 🐢

**Review by rahul-malik — APPROVED** [review]
Looks good, do we need to worry about cleaning up the indexer data at any point (`bazel clean`, etc.)?

File: src/TulsiGenerator/Scripts/bazel_build.py [hunk 1]
```diff
@@ -752,6 +758,20 @@ def _BuildBazelCommand(self, options):
 
     return (bazel_command, 0)
 
+  def _HydrateIndexStore(self):
```
Comment (jerrymarino) [inline src/TulsiGenerator/Scripts/bazel_build.py:761]: This method should check the Xcode version I think it will cause issues otherwise.

**Comment by jerrymarino** [comment]
@rahul-malik I left a comment inline - this PR was not 100% ready to merge..

**Pull request merged**